doc/sphinx-guides/source/developers/tools.rst

```diff
@@ -145,6 +145,139 @@ For example...
 
 would be consistent with a file descriptor leak on the dataset page.
 
+jmap and jstat
+++++++++++++++
+
+``jmap`` and ``jstat`` are parts of the standard JDK distribution. 
+jmap allows you to look at the contents of the java heap. It can be used to create a heap dump, that you can then feed to another tool, such as ``Memory Analyzer Tool`` (see above). It can also be used as a useful tool of its own, for example, to list all the classes currently instantiated in memory:
+
+.. code-block:: bash
+
+   $ jmap -histo <glassfish process id> 
+
+will output a list of all classes, sorted by the number of instances of each individual class, with the size in bytes. 
+This can be very useful when looking for memory leaks in the application. Another useful tool is ``jstat``, that can be used in combination with ``jmap`` to monitor the effectiveness of garbage collection in reclaiming allocated memory. 
+
+In the example script below we stress running Dataverse applicatione with GET requests to a specific dataverse page, use ``jmap`` to see how many Dataverse, Dataset and DataFile class object get allocated, then run ``jstat`` to see how the numbers are affected by both "Young Generation" and "Full" garbage collection runs (``YGC`` and ``FGC`` respectively):
+
+(This is script is provided **as an example only**! You will have to experiment and expand it to suit any specific needs and any specific problem you may be trying to diagnose, and this is just to give an idea of how to go about it)
+
+.. code-block:: bash
+
+   #!/bin/sh
```

Comment:
We could refactor this into a downloadable script and also continue to inline it. Here's an example of where we do that (the "clear timer" script): http://guides.dataverse.org/en/4.16/admin/troubleshooting.html#deployment-fails-ejb-timer-service-not-available